### PR TITLE
Nalexander/bug 962668 assertion skew

### PR DESCRIPTION
**Pull request by ncalexan**

### CODE REVIEW

File: src/main/java/org/mozilla/gecko/tokenserver/TokenServerClient.java [hunk 1]
```diff
@@ -203,11 +204,15 @@ public TokenServerToken processResponse(HttpResponse response)
...
   public void getTokenFromBrowserIDAssertion(final String assertion, final boolean conditionsAccepted,
       final TokenServerClientDelegate delegate) {
-    BaseResource r = new BaseResource(uri);
+    final BaseResource r = new BaseResource(uri);
 
     r.delegate = new BaseResourceDelegate(r) {
       @Override
       public void handleHttpResponse(HttpResponse response) {
+        SkewHandler skewHandler = SkewHandler.getSkewHandlerForResource(r);
+        if (skewHandler != null) {
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/tokenserver/TokenServerClient.java]: This should never be null, but that you're checking implies that you'd like it to be…?

File: src/main/java/org/mozilla/gecko/tokenserver/TokenServerClient.java [hunk 2]
```diff
@@ -203,11 +204,15 @@ public TokenServerToken processResponse(HttpResponse response)
...
+          skewHandler.updateSkew(response, System.currentTimeMillis());
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/tokenserver/TokenServerClient.java]: Ideally the wall clock time should be computed as early as possible, so lift `now` up to the start of the method.

(Example: what if fetching the skew handler causes a GC that takes 800ms?)

File: src/main/java/org/mozilla/gecko/tokenserver/TokenServerClient.java (hunk 1 above)
Comment (ncalexan) [inline src/main/java/org/mozilla/gecko/tokenserver/TokenServerClient.java]: Nah, just being safe.  I checked before part 2 and see it'll never be null, will update.

File: src/main/java/org/mozilla/gecko/tokenserver/TokenServerClient.java (hunk 2 above)
Comment (ncalexan) [inline src/main/java/org/mozilla/gecko/tokenserver/TokenServerClient.java]: Will do.

**Pull request merged**